Upgrade eslint

error-four-o-four opened the pull request:
adresses #1251 

1. Please make all pull requests on the `dev` branch ✅
2. Don't commit build files ✅
2. Try to get all [tests](https://github.com/Tonejs/Tone.js/wiki/Testing) to pass ✅

error-four-o-four committed:
chore: update linting scripts and dependencies

- Modified linting scripts in package.json to improve linting coverage and ease of use.
- Updated ESLint and related plugins to their latest versions.
- Removed unnecessary eslint-disable comments from various scripts.

error-four-o-four committed:
chore: introduce @stylistic ESLint plugins

error-four-o-four committed:
chore: update ESLint configuration and replace deprecate ts-rule 'empty interface'

error-four-o-four committed:
chore: update ESLint ignore patterns to include coverage, dist, and docs directories

tambien commented on test/scripts/test_integrations.mjs:
```diff
@@ -7,7 +7,6 @@ const integrations = await glob(resolve(__dirname, "../integration/*"));
 for (let dir of integrations) {
 	await within(async () => {
 		cd(dir);
-		// eslint-disable-next-line no-console
```
I would like to keep on no-console as a rule if possible. Sometimes when i'm debugging i use a lot of console.logs and would love to catch it before it's submitted.

error-four-o-four committed:
fix: apply elsint rules

@typescript-eslint/no-unused-expressions
@typescript-eslint/no-unused-vars

error-four-o-four committed:
refactor: update linting scripts to include TypeScript checks

- Modified the "lint" script to run TypeScript compiler with noEmit before ESLint.
- Updated "lint:fix" to include TypeScript checks and fix linting issues in both Tone and test directories.
- Removed unused ESLint plugin dependency "eslint-plugin-file-extension-in-import-ts".
